From f3cebd2cea644eef7a49438f0c261bda4fd9b8de Mon Sep 17 00:00:00 2001
From: Thang Tran <thuutran@amperecomputing.com>
Date: Wed, 2 Apr 2025 11:04:07 +0700
Subject: [PATCH] dbus-sdr: correct sensor type code of Power sensors

Issue: When users request to get the sensor properties via "ipmitool
sensor get <sensor_name>" command, the "Sensor Type" of Power sensors
are "Other".

Root cause: In the dynamic-sensor mode, the "Sensor Type" of a sensor is
indicated via the object path. Currently, the if the object path
includes "/power/" inside, the "Sensor Type" is "Other".

Solution: Update the table to map "power" to "Power Supply" type.

Tested:
   1. Get the power sensor via ipmitool commands
      ipmitool sensor get <power_sensor_name>
   2. The Sensor Type is "Power Supply"

Upstream-Status: Pending

Signed-off-by: Thang Tran <thuutran@amperecomputing.com>
---
 dbus-sdr/sdrutils.cpp         | 2 +-
 include/dbus-sdr/sdrutils.hpp | 1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/dbus-sdr/sdrutils.cpp b/dbus-sdr/sdrutils.cpp
index f6f6431..2ad86d3 100644
--- a/dbus-sdr/sdrutils.cpp
+++ b/dbus-sdr/sdrutils.cpp
@@ -55,7 +55,7 @@ boost::container::flat_map<
                                       SensorEventTypeCodes::sensorSpecified)},
          {"processor", std::make_pair(SensorTypeCodes::processor,
                                       SensorEventTypeCodes::sensorSpecified)},
-         {"power", std::make_pair(SensorTypeCodes::other,
+         {"power", std::make_pair(SensorTypeCodes::powerSupply,
                                   SensorEventTypeCodes::threshold)},
          {"memory", std::make_pair(SensorTypeCodes::memory,
                                    SensorEventTypeCodes::sensorSpecified)},
diff --git a/include/dbus-sdr/sdrutils.hpp b/include/dbus-sdr/sdrutils.hpp
index 9d36a50..61cccef 100644
--- a/include/dbus-sdr/sdrutils.hpp
+++ b/include/dbus-sdr/sdrutils.hpp
@@ -311,6 +311,7 @@ enum class SensorTypeCodes : uint8_t
     fan = 0x04,
     physicalSecurity = 0x5,
     processor = 0x07,
+    powerSupply = 0x08,
     powerUnit = 0x09,
     other = 0x0b,
     memory = 0x0c,
